cmd/frontend/graphqlbackend/graphqlbackend.go

```diff
@@ -427,8 +427,31 @@ func (r *NodeResolver) ToMonitorTriggerEvent() (MonitorTriggerEventResolver, boo
 	return n, ok
 }
 
-func (r *NodeResolver) ToCampaign() (CampaignResolver, bool) {
-	n, ok := r.Node.(CampaignResolver)
+// TODO(campaigns-deprecation): This should be removed once we remove campaigns completely
+func (r *NodeResolver) ToCampaign() (BatchChangeResolver, bool) {
+	n, ok := r.Node.(BatchChangeResolver)
+	if !ok {
+		return nil, false
+	}
+	return n, n.ActAsCampaign()
+}
+
+// TODO(campaigns-deprecation): This should be removed once we remove campaigns completely
+func (r *NodeResolver) ToCampaignSpec() (BatchSpecResolver, bool) {
+	n, ok := r.Node.(BatchSpecResolver)
+	if !ok {
+		return nil, false
+	}
+	return n, n.ActAsCampaignSpec()
+}
+
+func (r *NodeResolver) ToBatchChange() (BatchChangeResolver, bool) {
+	n, ok := r.Node.(BatchChangeResolver)
+	return n, ok
```

Comment:
@mrnugget won't this need to be so it can't ever flicker when graphql-go has no stable call order for these ToXX methods?
```suggestion
	return n, n.ActAsCampaignSpec()
```

Reply:
I don't think it's needed. graphql-go will only try to call `ToBatchSpec` if somebody queries `node(id: "foo") { ... on BatchSpec { id } }` and in that case it should be enough. But being explicit can't hurt I guess.

Reply:
```graphql
node(id: "foo") {
  __typename
}
```
I think this query would need to call toxx methods until it finds a match.

Reply:
Ah! I see. So if it's supposed to be a campaign then the `ToBatchSet()` could return true if it's called first.

Reply:
Yep, or worse the order maybe isn’t deterministic and sometimes it’s a campaign and sometimes it’s a batchchange😬